opendevin/core/utils/singleton.py

```diff
@@ -22,6 +22,9 @@ def reset(cls):
         for instance_type, instance in cls._instances.items():
             print('resetting... ', instance_type)
             for field in dataclasses.fields(instance_type):
+                if field.name == 'persist_sandbox':
+                    # do not reset the persist_sandbox flag
+                    continue
```

Comment:
```suggestion
```

Reply:
But current tests require clean sandboxes

Reply:
Ah, I see what happens. I made a PR to remove these resets from the sandboxes tests. It works both on local and on CI, so it really should not be there at all.

Then the env var you set in this PR should be respected.